go-controller/pkg/ovn/ovn.go

```diff
@@ -609,6 +638,105 @@ func (oc *Controller) GetServiceVIPToName(vip string, protocol kapi.Protocol) (t
 	return namespace, ok
 }
 
+// AddServiceLBToACL associates an empty load balancer with its associated ACL reject rule
+func (oc *Controller) SetServiceACLToLB(lb, vip, acl string) {
+	oc.serviceLBLock.Lock()
+	defer oc.serviceLBLock.Unlock()
+	if _, ok := oc.serviceLBMap[lb]; !ok {
+		oc.serviceLBMap[lb] = make(map[string]*LoadBalancerConf)
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{RejectACL: acl}
+		return
+	}
+	if _, ok := oc.serviceLBMap[lb][vip]; !ok {
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{RejectACL: acl}
+		return
+	}
+	// This should not happen, the entry should be removed for the VIP when the service is deleted for an existing
+	// entry
+	if oc.serviceLBMap[lb][vip] == nil {
+		klog.Errorf("config for existing load balancer is nil. LB: %s, VIP: %s", lb, vip)
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{RejectACL: acl}
+		return
+	}
+	oc.serviceLBMap[lb][vip].RejectACL = acl
+}
+
+// SetServiceEndpointsToLB associates a load balancer with endpoints
+func (oc *Controller) SetServiceEndpointsToLB(lb, vip string, eps []string) {
+	oc.serviceLBLock.Lock()
+	defer oc.serviceLBLock.Unlock()
+	if _, ok := oc.serviceLBMap[lb]; !ok {
+		oc.serviceLBMap[lb] = make(map[string]*LoadBalancerConf)
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{Endpoints: eps}
+		return
+	}
+	if _, ok := oc.serviceLBMap[lb][vip]; !ok {
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{Endpoints: eps}
+		return
+	}
+	// This should not happen, the entry should be removed for the VIP when the service is deleted
+	if oc.serviceLBMap[lb][vip] == nil {
+		klog.Errorf("config for existing load balancer is nil. LB: %s, VIP: %s", lb, vip)
+		oc.serviceLBMap[lb][vip] = &LoadBalancerConf{Endpoints: eps}
+		return
+	}
+	oc.serviceLBMap[lb][vip].Endpoints = eps
+}
+
+func (oc *Controller) GetServiceLBACL(lb, vip string) (string, bool) {
+	if conf, ok := oc.GetServiceLBConfig(lb, vip); ok {
```

Comment:
@trozet just do `...; ok && conf.RejectACL != "" {` and then callers don't need to check for `!ok || acl == ""`

Reply:
I can do that, then there is no reason to return 2 vars (string and bool) and just return the string. I think when I wrote this I did it this way in case there was a need for a caller to distinguish between:
A) does the entry even exist?
B) the entry exists but is just empty

I guess the code calling it could care less, so might as well just make it like you suggested. Note, GetServiceLBEndpoints operates in the same fashion. The caller checks:
if eps, ok := ovn.GetServiceLBEndpoints(loadBalancer, vip); ok && len(eps) > 0 {

I'm guessing you want me to do the same thing for that one too right?

Reply:
@trozet yeah same there. If GetServiceLBConfig() is always going to create an empty struct when it doesn't yet exist, and return that struct, I don't think there's a point to having 'ok' at this time.